File: modules/jpa/testdata/java/reference/model/BaseEntity.java
package com.example;

import javax.persistence.*;
import jakarta.persistence.*;
import java.util.List;

@MappedSuperclass
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name = "entity_type")
public abstract class BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }
}

File: modules/jpa/testdata/kotlin/reference/model/BaseEntity.kt
package com.example

import jakarta.persistence.*
import javax.persistence.*

@MappedSuperclass
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name = "entity_type")
abstract class BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
}